components/Registration/Registration.tsx

```diff
@@ -86,14 +90,170 @@ const RegistrationForm: React.FC = () => {
+        fieldValues: FieldValues
+    ): Promise<RegistrationType> => {
+        const registrationWithId = await getRegistration();
+        let finalRegistrationObject: RegistrationType = {
+            preferredName: "",
+            legalName: "",
+            emailAddress: "",
+            university: "",
+            hackEssay1: "",
+            hackEssay2: "",
+            optionalEssay: "",
+            resumeFileName: "",
+            location: "",
+            gender: "",
+            degree: "",
+            major: "",
+            minor: "",
+            gradYear: 0,
+            isProApplicant: false,
+            proEssay: "",
+            considerForGeneral: false,
+            requestedTravelReimbursement: false,
+            dietaryRestrictions: [],
+            race: [],
+            hackInterest: [],
+            hackOutreach: []
+        };
+
+        if (registrationWithId) {
+            const { id, ...registration } = registrationWithId;
+            finalRegistrationObject = {
+                ...finalRegistrationObject,
+                ...registration
+            };
+        }
+
+        for (const key in finalRegistrationObject) {
+            if (key in fieldValues) {
+                if (key === "race") {
+                    const raceValue = fieldValues[key as keyof FieldValues];
+                    finalRegistrationObject.race = raceValue
+                        ? Array.isArray(raceValue)
+                            ? (raceValue as string[])
+                            : [raceValue as string]
+                        : [];
+                } else if (key === "gradYear") {
+                    const gradYearValue = fieldValues[key as keyof FieldValues];
+                    finalRegistrationObject.gradYear = Number(gradYearValue);
+                } else if (key === "requestedTravelReimbursement") {
+                    if (fieldValues[key as keyof FieldValues][0] === "YES") {
+                        finalRegistrationObject.requestedTravelReimbursement =
+                            true;
+                    }
+                } else {
+                    // console.log("key", key);
+                    finalRegistrationObject[key as keyof RegistrationType] =
+                        fieldValues[key as keyof FieldValues];
+                }
+            }
+        }
+
+        return finalRegistrationObject;
+    };
+
+    const onSubmit = async (
         values: FieldValues,
         formikHelpers: FormikHelpers<FieldValues>
     ) => {
-        console.log("submit", values);
+        // console.log("submit", values);
+
+        const registrationObject = await createRegistrationObject(values);
+
+        // console.log("full thing", registrationObject);
+
         handlePageChange(formIndex + 1);
+
+        registerUpdate(registrationObject)
+            // repsonse is not returning resumefileName even though registration object has the field, can't figure out why
```

Comment:
It seems like the resume field is handled by making requests to /s3/upload and /s3/download/{id}. https://adonix.hackillinois.org/docs/#/

Reply:
The upload might need to be done in the file upload component every time the user selects a new file and not when the next button is pressed